cpp/include/cuvs/neighbors/common_defs.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file common_defs.h
```

Comment:
Hi @benfred @cjnolet, I introduced a new C header to hold the declaration shared between C & C++ domains, while minimizing the introduction of extra headers and C objects. Could you take a moment to help review? Thanks!

Reply:
Any reason these can't be added to `common.h`?

Reply:
When we added these defines to common.h, when including it in cagra.hpp, many unnecessary symbols will dirty the C++ global space, since there are many headers in common.h and two enums with not-unique member naming.
The more difficult thing is that it will cause compilation error for not found '#include <dlpack/dlpack.h>'

Reply:
Hello @benfred @cjnolet , I did some changes on the headers arrangements, would you please take a moment to review? Thanks!